flink-clients/src/main/java/org/apache/flink/client/ClientUtils.java

```diff
@@ -80,4 +93,43 @@ public static ClassLoader buildUserCodeClassLoader(List<URL> jars, List<URL> cla
 		}
 		return FlinkUserCodeClassLoaders.parentFirst(urls, parent);
 	}
+
+	public static JobSubmissionResult executeProgram(
```

Comment:
to exclude detach configuration from ClusterClient, it would be later temporary move as a parameter of this method. This method is a temporary workaround to finish the job done by `ContextEnvironment`. I think FLIP-73 would properly handle it.